### PR TITLE
[REVIEW] fix JNI gcc-9 compile error [skip ci]

### PR DESCRIPTION
**Pull request by rongou**
Make the JNI code compile under gcc-9 with CUDA 11.

The file got clang-formatted, the only real change is the last one.

@jlowe @revans2

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: java/src/main/native/src/ScalarJni.cpp [hunk 1]
```diff
@@ -321,7 +334,7 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_Scalar_makeStringScalar(JNIEnv *env,
       strval.assign(reinterpret_cast<char *>(jbytes.data()), jbytes.size());
     }
 
-    auto s = new cudf::string_scalar{strval, is_valid};
+    auto s = new cudf::string_scalar{strval, static_cast<bool>(is_valid)};
```
Comment (jlowe) [inline java/src/main/native/src/ScalarJni.cpp:337]: Curious, why do we need to static cast a `jboolean` parameter here but not in other cases like line 317 of this file or [here](https://github.com/rapidsai/cudf/blob/branch-0.15/java/src/main/native/src/TableJni.cpp#L1726).

Comment (rongou) [reply java/src/main/native/src/ScalarJni.cpp:337]: The curly brace initialization (aka list initialization) does less implicit type conversion. I guess gcc-9 is more strict about it now.

**Pull request merged**